### ISSUE TITLE
LibJS: !value.is_empty() assertion failure in Object::put

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#6642]
Found with Fuzzilli.

Code:
```js
function main() {
const v3 = "number".slice(1819080160);
const v5 = [1337,1337];
const v6 = {__proto__:v5,a:v5,length:1819080160};
const v7 = eval(v3);
for (const v8 in v6) {
}
}
main();
// STDERR:
// FuzzilliJs: ../Userland/Libraries/LibJS/Runtime/Object.cpp:833: virtual bool JS::Object::put(const JS::PropertyName &, JS::Value, JS::Value): Assertion `!value.is_empty()' failed.
```

Trace:
```
js(28): USERSPACE(28) ASSERTION FAILED: !value.is_empty()
../../Userland/Libraries/LibJS/Runtime/Object.cpp:833
CrashDaemon(15): --- Backtrace for thread #0 (TID 28) ---
CrashDaemon(15): 0x778ff1ce: [libsystem.so] syscall2 +0xe (syscall.cpp:25)
CrashDaemon(15): 0x4c90e8b0: [libc.so] kill +0x20 (signal.cpp:21)
CrashDaemon(15): 0x4c90e953: [libc.so] raise +0x23 (signal.cpp:34)
CrashDaemon(15): 0x4c8fc25d: [libc.so] abort +0x2a (stdlib.cpp:232)
CrashDaemon(15): 0x4c8fcd4b: [libc.so] __assertion_failed +0xeb (Format.h:236)
CrashDaemon(15): 0xae82c99f: [libjs.so] JS::Object::put(JS::PropertyName const&, JS::Value, JS::Value) +0x2cf (StringOrSymbol.h:161)
CrashDaemon(15): 0xae876bda: [libjs.so] JS::VM::set_variable(AK::FlyString const&, JS::Value, JS::GlobalObject&, bool) +0x12a (Atomic.h:246)
CrashDaemon(15): 0xae782e9f: [libjs.so] JS::VariableDeclaration::execute(JS::Interpreter&, JS::GlobalObject&) const +0x1af (AST.cpp:1539)
CrashDaemon(15): 0xae79d6d5: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0xa5 (Interpreter.cpp:149)
CrashDaemon(15): 0xae84fb0f: [libjs.so] JS::ScriptFunction::execute_function_body() +0x2ef (ScriptFunction.cpp:130)
CrashDaemon(15): 0xae84fe8d: [libjs.so] JS::ScriptFunction::call() +0x16d (ScriptFunction.cpp:139)
CrashDaemon(15): 0xae87896b: [libjs.so] JS::VM::call_internal(JS::Function&, JS::Value, AK::Optional<JS::MarkedValueList>) +0x78b (Vector.h:156)
CrashDaemon(15): 0xae787f2d: [libjs.so] JS::CallExpression::execute(JS::Interpreter&, JS::GlobalObject&) const +0xe2d (VM.h:274)
CrashDaemon(15): 0xae780c9f: [libjs.so] JS::ExpressionStatement::execute(JS::Interpreter&, JS::GlobalObject&) const +0xaf (AST.cpp:50)
CrashDaemon(15): 0xae79d6d5: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0xa5 (Interpreter.cpp:149)
CrashDaemon(15): 0xae7808bc: [libjs.so] JS::Program::execute(JS::Interpreter&, JS::GlobalObject&) const +0xac (AST.cpp:50)
CrashDaemon(15): 0xae79c49b: [libjs.so] JS::Interpreter::run(JS::GlobalObject&, JS::Program const&) +0x28b (VM.h:154)
CrashDaemon(15): 0x02408cbd: [/bin/js] parse_and_run(JS::Interpreter&, AK::StringView const&) +0x11d (js.cpp:484)
CrashDaemon(15): 0x0240118b: [/bin/js] main +0xb8b (js.cpp:985)
CrashDaemon(15): 0x024019d8: [/bin/js] _start +0x58 (crt0.cpp:39)
```

**Comment by linusg** [comment]
Simplified repro:

```js
a = eval("");
```

**Comment by linusg** [comment]
Fixed in 7b1ba4bd5cdc2df2e83253e5a7935de70b4ab338.
